enterprise_access/apps/content_assignments/management/commands/automatically_expire_assignments.py

```diff
@@ -0,0 +1,131 @@
+            )
+
+            allocated_assignments = assignment_configuration.assignments.filter(
+                state=LearnerContentAssignmentStateChoices.ALLOCATED
+            )
+
+            paginator = Paginator(allocated_assignments, 100)
+            for page_number in paginator.page_range:
+                assignments = paginator.page(page_number)
+
+                content_metadata_for_assignments = get_content_metadata_for_assignments(
+                    enterprise_catalog_uuid,
+                    assignments
+                )
+
+                for assignment in assignments:
+                    content_metadata = content_metadata_for_assignments.get(assignment.content_key, {})
+                    enrollment_end_date = content_metadata.get('normalized_metadata', {}).get('enroll_by_date')
+
+                    logger.info(
+                        '%s AssignmentUUID: [%s], ContentKey: [%s], AssignmentExpiry: [%s], EnrollmentEnd: [%s], SubsidyExpiry: [%s]',  # nopep8 pylint: disable=line-too-long
+                        log_prefix,
+                        assignment.uuid,
+                        assignment.content_key,
+                        assignment.created + timezone.timedelta(days=90),
+                        enrollment_end_date,
+                        subsidy_expiration_datetime
+                    )
+
+                    expired_assignment_uuid = None
+                    # just for logging
+                    assignment_expiry_reason = None
+
+                    if current_date > (assignment.created + timezone.timedelta(days=90)):
+                        expired_assignment_uuid = assignment.uuid
+                        assignment_expiry_reason = AssignmentAutomaticExpiredReason.NIENTY_DAYS_PASSED
+                    elif enrollment_end_date and enrollment_end_date < current_date:
+                        expired_assignment_uuid = assignment.uuid
+                        assignment_expiry_reason = AssignmentAutomaticExpiredReason.ENROLLMENT_DATE_PASSED
+                    elif subsidy_expiration_datetime and subsidy_expiration_datetime < current_date:
+                        expired_assignment_uuid = assignment.uuid
+                        assignment_expiry_reason = AssignmentAutomaticExpiredReason.SUBSIDY_EXPIRED
+
+                    if expired_assignment_uuid:
+                        logger.info(
+                            '%s Assignment Expired. UUID: [%s], Reason: [%s]',
+                            log_prefix,
+                            expired_assignment_uuid,
+                            assignment_expiry_reason
+                        )
+
+                        if not dry_run:
+                            assignment.state = LearnerContentAssignmentStateChoices.CANCELLED
+                            assignment.save()
+                            send_assignment_automatically_expired_email.delay(expired_assignment_uuid, admin_emails)
+
+                sleep(sleep_duration)
```

Comment:
just curious - what's the `sleep()` for?

Reply:
My understanding was to wait for some time before processing the next batch of assignments? The whole purpose was to let as many celery tasks as possible to complete their work? Thoughts?

Reply:
Ah, I see.  The celery broker can handle the queueing of tasks for us, I don't think we need to `sleep()` here.